### PR TITLE
test: add rpc_fundrawtransaction logging

### PR DESCRIPTION
**Pull request by jonatack**
`test/functional/rpc_fundrawtransaction.py` is fairly slow to run and has no logging, so it can appear to be stalled.

This commit adds info logging at each test to provide feedback on the test run.

### CODE REVIEW

**Comment by jonatack** [comment]
Motivation: Watching @achow101 swear repeatedly at this test while debugging CoinSelection changes because it looked like the test was stalled :)

**Comment by instagibbs** [comment]
since you're touching so many lines, maybe just take the ASCII art headers and turn the actual text into the print line? sorry :grimacing:

File: test/functional/rpc_fundrawtransaction.py [hunk 1]
```diff
@@ -305,6 +318,7 @@ def test_two_vin(self):
         ###########################################
         # test a fundrawtransaction with two VINs #
```
Comment (instagibbs) [inline test/functional/rpc_fundrawtransaction.py]: f.e. just convert this line into a print statement

Comment (MarcoFalke) [reply test/functional/rpc_fundrawtransaction.py]: Jup, can remove this now

File: test/functional/rpc_fundrawtransaction.py [hunk 2]
```diff
@@ -41,6 +41,7 @@ def setup_network(self):
         connect_nodes(self.nodes[0], 3)
 
     def run_test(self):
+        self.log.info("Setup, generating 126 nodes and syncing...")
```
Comment (jnewbery) [inline test/functional/rpc_fundrawtransaction.py]: Do you mean 126 blocks? Also, I think there are only 122. Perhaps best to just omit the number?

File: test/functional/rpc_fundrawtransaction.py [hunk 3]
```diff
@@ -464,6 +499,7 @@ def test_fee_4of5(self):
     def test_spend_2of2(self):
         ############################################################
         # spend a 2of2 multisig transaction over fundraw
+        self.log.info("Testing fundrawtxn spending 2of2 multisig...")
```
Comment (jnewbery) [inline test/functional/rpc_fundrawtransaction.py]: nit: prefer 2-of-2

File: test/functional/rpc_fundrawtransaction.py [hunk 4]
```diff
@@ -695,6 +739,7 @@ def test_address_reuse(self):
         ################################
         # Test no address reuse occurs #
         ################################
+        self.log.info("Testing fundrawtxn no address reuse occurs")
```
Comment (jnewbery) [inline test/functional/rpc_fundrawtransaction.py]: this sentence sounds a bit clunky. Perhaps "Test no address reuse occurs with `fundrawransaction`"

**Review by jnewbery — COMMENTED** [review]
Thanks for doing this. A few personal style nits:

- I think the indicative would sound better in these test logs (eg "test x" rather than "testing x")
- agree with the other reviewers that the logs can replace the comment blocks
- I'd prefer to remove the trailing '...' ellipses. It's not standard that they mean "this takes a long time"

File: test/functional/rpc_fundrawtransaction.py (hunk 2 above)
Comment (jonatack) [reply test/functional/rpc_fundrawtransaction.py]: Thanks @jnewbery! 4 nodes and 122 blocks. Replacing with "Connect nodes, set fees, generate blocks, and sync"

File: test/functional/rpc_fundrawtransaction.py (hunk 1 above)
Comment (jonatack) [reply test/functional/rpc_fundrawtransaction.py]: Done, ASCII art blocks removed.

File: test/functional/rpc_fundrawtransaction.py (hunk 3 above)
Comment (jonatack) [reply test/functional/rpc_fundrawtransaction.py]: Done, did the same for "4of5"

File: test/functional/rpc_fundrawtransaction.py (hunk 4 above)
Comment (jonatack) [reply test/functional/rpc_fundrawtransaction.py]: Agree, replacing with "Test fundrawtxn does not reuse addresses" to maintain consistent verb + op noun logging sentences.

**Comment by laanwj** [comment]
Concept ACK, agree with @jnewbery's nits.

**Review by jonatack — COMMENTED** [review]
Thank you @instagibbs, @MarcoFalke, and @jnewbery for reviewing, updated with your suggestions.

**Comment by jonatack** [comment]
Test output after updates:
```
$ test/functional/rpc_fundrawtransaction.py 
2019-11-01T10:58:43.324000Z TestFramework (INFO): Initializing test directory /tmp/bitcoin_func_test_vh0ej7_v
2019-11-01T10:58:44.504000Z TestFramework (INFO): Connect nodes, set fees, generate blocks, and sync
2019-11-01T10:58:46.250000Z TestFramework (INFO): Test fundrawtxn changePosition option
2019-11-01T10:58:47.847000Z TestFramework (INFO): Test fundrawtxn
2019-11-01T10:58:47.950000Z TestFramework (INFO): Test fundrawtxn with 2 coins
2019-11-01T10:58:48.025000Z TestFramework (INFO): Test fundrawtxn with 2 outputs
2019-11-01T10:58:48.093000Z TestFramework (INFO): Test fundrawtxn with a vin > required amount
2019-11-01T10:58:48.150000Z TestFramework (INFO): Test fundrawtxn not having a change output
2019-11-01T10:58:48.209000Z TestFramework (INFO): Test fundrawtxn with an invalid option
2019-11-01T10:58:48.241000Z TestFramework (INFO): Test fundrawtxn with an invalid change address
2019-11-01T10:58:48.274000Z TestFramework (INFO): Test fundrawtxn with a provided change address
2019-11-01T10:58:48.331000Z TestFramework (INFO): Test fundrawtxn with a provided change type
2019-11-01T10:58:48.382000Z TestFramework (INFO): Test fundrawtxn with a vin < required amount
2019-11-01T10:58:48.493000Z TestFramework (INFO): Test fundrawtxn with 2 vins
2019-11-01T10:58:48.645000Z TestFramework (INFO): Test fundrawtxn with 2 vins and 2 vouts
2019-11-01T10:58:48.804000Z TestFramework (INFO): Test fundrawtxn with an invalid vin
2019-11-01T10:58:48.889000Z TestFramework (INFO): Test fundrawtxn p2pkh fee
2019-11-01T10:58:49.005000Z TestFramework (INFO): Test fundrawtxn p2pkh fee with multiple outputs
2019-11-01T10:58:49.540000Z TestFramework (INFO): Test fundrawtxn fee with 4-of-5 addresses
2019-11-01T10:58:49.775000Z TestFramework (INFO): Test fundrawtxn spending 2-of-2 multisig
2019-11-01T10:59:03.159000Z TestFramework (INFO): Test fundrawtxn with locked wallet
2019-11-01T10:59:06.770000Z TestFramework (INFO): Test fundrawtxn fee with many inputs
2019-11-01T10:59:20.289000Z TestFramework (INFO): Test fundrawtxn sign+send with many inputs
2019-11-01T10:59:38.379000Z TestFramework (INFO): Test fundrawtxn with OP_RETURN and no vin
2019-11-01T10:59:38.415000Z TestFramework (INFO): Test fundrawtxn using only watchonly
2019-11-01T10:59:38.614000Z TestFramework (INFO): Test fundrawtxn using entirety of watched funds
2019-11-01T10:59:38.769000Z TestFramework (INFO): Test fundrawtxn feeRate option
2019-11-01T10:59:39.215000Z TestFramework (INFO): Test fundrawtxn does not reuse addresses
2019-11-01T10:59:39.382000Z TestFramework (INFO): Test fundrawtxn subtractFeeFromOutputs option
2019-11-01T10:59:40.115000Z TestFramework (INFO): Stopping nodes
2019-11-01T10:59:40.385000Z TestFramework (INFO): Cleaning up /tmp/bitcoin_func_test_vh0ej7_v on exit
2019-11-01T10:59:40.385000Z TestFramework (INFO): Tests successful
```

**Comment by instagibbs** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/17327/commits/ff22751417c6fbbd22f4eefd0e23431a83335c13

**Comment by jnewbery** [comment]
tACK ff22751417c6fbbd22f4eefd0e23431a83335c13

**Pull request merged**